### PR TITLE
Add auto-merge

### PR DESCRIPTION
**Pull request by WilliamBZA**

### CODE REVIEW

File: .github/dependabot-auto-merge.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+name: Dependabot Auto Merge
```
Comment (adamralph) [inline .github/dependabot-auto-merge.yml:1]: This file has to be in `.github/workflows/`.

File: .github/dependabot-auto-merge.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  pull_request:
+jobs:
+  auto-merge:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ahmadnassri/action-dependabot-auto-merge@v2
+        with:
+          target: minor
+          github-token: ${{ secrets.DependabotAutoMerge }}
```
Comment (adamralph) [inline .github/dependabot-auto-merge.yml:12]: The naming convention for secrets is UPPER_SLUG_CASE, and I think it should be explicit that this is a token.

```suggestion
          github-token: ${{ secrets.DEPENDABOT_AUTO_MERGE_TOKEN }}
```

File: .github/auto-merge.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+- match:
+    dependency_name: Microsoft.AspNetCore.Server.Kestrel
+    update_type: all
+
+- match:
+    dependency_name: Microsoft.AspNetCore.StaticFiles
+    update_type: all
+
+- match:
+    dependency_name: Microsoft.CodeAnalysis.CSharp.Workspaces
```
Comment (adamralph) [inline .github/auto-merge.yml:22]: We can't automerge this one. We need to stay on an older version to ensure the analyzers built into the NSB package can [support older versions of VS](https://docs.microsoft.com/en-us/visualstudio/extensibility/roslyn-version-support?view=vs-2019).

File: .github/auto-merge.yml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+- match:
+    dependency_name: Moq
+    update_type: all
+
+- match:
+    dependency_name: NETStandard.Library
+    update_type: all
+
+- match:
+    dependency_name: NUnit
```
Comment (adamralph) [inline .github/auto-merge.yml:46]: I guess we can probably automerge this one, but we have to bear in mind that it is a public dependency of some of our testing packages. E.g. https://www.nuget.org/packages/NServiceBus.AcceptanceTesting/

Comment (WilliamBZA) [reply .github/auto-merge.yml:46]: Is it possible to overwrite the repostandards files in a specific repo? If yes, we can exclude it from that specific repo by using a custom config

Comment (WojcikMike) [reply .github/auto-merge.yml:70]: @WilliamBZA yes it is: https://github.com/Particular/NServiceBus.Serializers.CompatTests/pull/32/files
So in a given repo we can specify that a config file will not be pulled in

Comment (adamralph) [reply .github/auto-merge.yml:70]: The problem with that is that you exclude the _whole_ file, so any other updates to the file will have to be manually propagated to any repo which is excluding that file.

Comment (WojcikMike) [reply .github/auto-merge.yml:70]: yep. The whole question that William asks is a bit odd. What I mean by that: dependabot.yml is a file that lives in a repo, so every repository defines a list of dependencies that are allowed and forbidden from PR's being raised. However, as we push this file using RepoStandards we lead ourselves into having one global list of dependencies.

So far if we will have a dependency that could be auto-merged in some repos and not in the others I would vote to simply leave it out of auto-merging. The list that we have, should already make the dependabot more manageable.

Comment (adamralph) [reply .github/auto-merge.yml:70]: > dependabot.yml is a file that lives in a repo, so every repository defines a list of dependencies that are allowed and forbidden from PR's being raised

`dependabot.yml` is also [defined in this repo](https://github.com/Particular/RepoStandards/blob/master/.github/dependabot.yml), so every repo has exactly the same file. It does not have an inclusion or exclusion list of dependencies. Dependabot raises for PR's for any dependency it is able to do so for.

> So far if we will have a dependency that could be auto-merged in some repos and not in the others I would vote to simply leave it out of auto-merging. The list that we have, should already make the dependabot more manageable.

I agree.

File: .github/auto-merge.yml [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+- match:
+    dependency_name: Mono.Cecil
+    update_type: all
+
+- match:
+    dependency_name: Moq
+    update_type: all
+
+- match:
+    dependency_name: NETStandard.Library
```
Comment (adamralph) [inline .github/auto-merge.yml:62]: This one is an edge case, and I would probably remove it. Having an update to this dep would be quite a weird thing, and I think if it ever happened, we want to verify why, and if we want it.

**Review by WojcikMike — APPROVED** [review]

**Pull request merged**